week3/word.cpp

```diff
@@ -0,0 +1,18 @@
+//
+// Created by julia on 6/8/17.
+//
+
+#include "word.h"
```

Comment:
I decided that it isn't important whether there is a dictionary class or a word class because the structures are similar. But when the names of methods are is_word and is_prefix, it's more habitual for me to think of it like the class object is word or prefix. So I prefer construction word.is_word(set) to Dictionary.is_word(word).